src/lib.rs

```diff
@@ -1,748 +1,268 @@
-    get_entry_point_function_name:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> *const u8>,
-    needs_function_constants:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> bool>,
-    get_function_constant_count:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> usize>,
-
-    copy_function_constants:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    release_function_constants:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    copy_compute_info: libloading::Symbol<
-        'lib,
-        unsafe extern "C" fn(
-            *mut IRShaderReflectionOpaque,
-            version: IRReflectionVersion,
-            csinfo: *mut IRVersionedCSInfo,
-        ) -> bool,
-    >,
-    copy_vertex_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    copy_fragment_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    copy_geometry_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    copy_hull_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    copy_domain_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    release_compute_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    release_vertex_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    release_fragment_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    release_geometry_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    release_hull_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    release_domain_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    get_resource_count:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    get_resource_locations:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-}
-
-pub struct IRShaderReflection<'lib> {
-    me: *mut IRShaderReflectionOpaque,
-    funcs: IRReflectionFn<'lib>,
-}
-
-impl<'lib> Drop for IRShaderReflection<'lib> {
+use std::{
+    ffi::{CStr, OsStr},
+    mem::MaybeUninit,
+    sync::Arc,
+};
+
+#[allow(
+    non_upper_case_globals,
+    dead_code,
+    non_camel_case_types,
+    non_snake_case,
+    clippy::too_many_arguments,
+    clippy::enum_variant_names
+)]
+mod bindings;
+
+pub use bindings::{
```

Comment:
So as final cleanup, let's replace this with `pub use bindings as ffi;` and prepend every non-high-level thing with `ffi::`.